security/fma-interop-access-list.md

```diff
@@ -0,0 +1,121 @@
+- **Recovery Path(s):** Deploy contract upgrade with adjusted thresholds if gas schedule changes significantly.
+
+### FM2: Storage Slot Collision
+
+- **Description:** The checksum calculation for message validation uses storage slots that could potentially collide with other slots from the access list, leading to false validation of messages.
+- **Risk Assessment:** High
+  - Potential Impact: Critical. Storage collisions could allow unauthorized messages to be validated.
+  - Likelihood: Low. Storage slots are carefully calculated using Identifier and message hash parameter. The way both are encoded in the contract’s logic needs to be flawed for this case to happen.
+- **Mitigations**
+  - Implement robust checksum calculation that minimizes collision risk
+  - Add tests verifying no collisions occur with expected storage patterns
+  - Document storage layout and slot calculation methodology
+- **Detection:** Monitor for unexpected message validations that could indicate storage collisions.
+- **Recovery Path(s):** Deploy contract upgrade with revised storage slot calculation if collisions are detected and update ALL off-chain tools and SDKs that generate these hashes on the access list.
+
+### FM3: Access List Manipulation
+
+- **Description:** Malicious actors could attempt to manipulate transaction access lists to force validation of unauthorized messages.
+- **Risk Assessment:** High
+  - Potential Impact: High. Could allow unauthorized message execution.
+  - Likelihood: Medium. Access lists are user-controlled transaction parameters.
+- **Mitigations:**
+  - Ensure checksum calculation is cryptographically sound
+  - Implement proper validation of access list entries
+  - Add checks for maximum access list size
+- **Detection:** Monitor for unusual patterns in access list usage and message validation attempts.
+- **Recovery Path(s):** Pause message processing if manipulation is detected. Deploy fixes for validation logic.
+
+### FM4: Relayer Implementation Errors
+
+- **Description:** Relayers may implement access list construction incorrectly, leading to failed message validation even for valid messages. One possible cause for this could be the checksum calculation not matching between clients and on-chain logic.
+- **Risk Assessment:** Medium
+  - Potential Impact: Medium. Could cause temporary disruption of cross-chain messaging.
+  - Likelihood: Medium. Relayers need to adopt new patterns for access list construction.
+- **Mitigations:**
+  - Provide clear documentation and examples for relayer implementations
+  - Create reference implementations and test suites
+  - Add monitoring for failed validation attempts
+- **Detection:** Track validation failure rates and patterns across relayers.
+- **Recovery Path(s):** Work with relayer implementations to fix access list construction.
+
+### FM5: Compiler Optimizations Break Gas Introspection
```

Comment:
This is a very good failure mode. I think that e2e testing should be able to catch this, if we have an e2e test that asserts the revert. Doing it as an e2e test ensures nothing about the foundry environment impacts the gas consumption